service/datastore/api/src/main/java/org/eclipse/kapua/service/datastore/model/query/SortField.java

```diff
@@ -8,43 +8,49 @@
  *
  * Contributors:
  *     Eurotech - initial API and implementation
+ *     Red Hat Inc
  *******************************************************************************/
 package org.eclipse.kapua.service.datastore.model.query;
 
 /**
  * Sortable field definition
- * 
- * @since 1.0
  *
  */
-public interface SortField {
-
-    /**
-     * Get the field name
-     * 
-     * @return
-     */
-    public String getField();
-
-    /**
-     * Set the field name
-     * 
-     * @param field
-     */
-    public void setField(String field);
-
-    /**
-     * Get the sort direction
-     * 
-     * @return
-     */
-    public SortDirection getSortDirection();
-
-    /**
-     * Set the sort direction
-     * 
-     * @param sortDirection
-     */
-    public void setSortDirection(SortDirection sortDirection);
+public class SortField {
 
+    private SortDirection sortDirection;
+    private String field;
+
+    private SortField(final SortDirection direction, final String field) {
+        this.sortDirection = direction;
+        this.field = field;
+    }
+
+    public String getField() {
+        return this.field;
+    }
+
+    public SortDirection getSortDirection() {
+        return this.sortDirection;
+    }
+
+    public static SortField of(SortDirection direction, final String fieldName) {
+        if (fieldName == null || fieldName.isEmpty()) {
+            return null;
```

Comment:
Well this method could work with null values, by the way provided. So there is no real need to return a NPE. However the method receiving the "SortField" instance, may or may not be capable of handling null, so I would say let this method then handle the null situation.

Assume you have some UI or API which allows you to omit the sort field, then you would have a null check in this method and the method receiving the SortField instance. Letting the receiver handle the null situation alone would limit the null-check to the methods which really have a problem receiving null.